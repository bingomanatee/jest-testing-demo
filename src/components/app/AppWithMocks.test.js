import {clear} from "@testing-library/user-event/dist/clear";
import {cleanup, render} from "@testing-library/react";

describe('App', () => {
  describe('snapshot', () => {
    describe('will succeed', () => {
      let LocalApp;

      beforeEach(async () => {
        jest.mock('./timestamp', () => {
          return {timestamp: () => '___ mock timestamp ___'}
        });

        const appModule = await import('./App');
        LocalApp = function LocalApp() {
          const BaseApp = appModule.default;
          return <div>
            <BaseApp />
          </div>
        }
        console.log('local app is ', LocalApp);
      });

      afterEach(() => {
        cleanup();
      });

      it('should render a snapshot of the store with mock date', () => {
        const { baseElement } = render(<LocalApp />);
        expect(baseElement).toMatchSnapshot();
      })
    })

  })

})