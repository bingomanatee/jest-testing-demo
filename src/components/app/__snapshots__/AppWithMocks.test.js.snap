// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot will succeed should render a snapshot of the store with mock date 1`] = `
<body>
  <div>
    <div>
      <div
        class="App"
      >
        <header
          class="App-header"
        >
          <img
            alt="logo"
            class="App-logo"
            src="logo.svg"
          />
          <p>
            The Finfare T Shirt Online Store
          </p>
        </header>
        <div
          class="product"
        >
          <h3>
            Mens Shirt
          </h3>
          <p>
            A hefty manly man's shirt
          </p>
          <button
            type="button"
          >
            Add to Cart
          </button>
        </div>
        <div
          class="product"
        >
          <h3>
            Womens Shirt
          </h3>
          <p>
            A hefty womanly woman's shirt
          </p>
          <button
            type="button"
          >
            Add to Cart
          </button>
        </div>
        <section>
          <hr />
          <h2>
            Shopping cart 
            ___ mock timestamp ___
          </h2>
          <ul />
        </section>
      </div>
    </div>
  </div>
</body>
`;
